src/state/zustand/swapperStore/amountSelectors.ts

```diff
@@ -357,3 +423,144 @@ export const selectTradeAmountsByActionAndAmount: Selector<
+
+    const quoteBuyAmountCryptoBaseUnit = toBaseUnit(
+      quoteBuyAmountCryptoPrecision,
+      buyAssetPrecision,
+    )
+    return bnOrZero(quoteBuyAmountCryptoBaseUnit)
+      .minus(buyAssetTradeFeeBuyAssetBaseUnit)
+      .minus(sellAssetTradeFeeBuyAssetBaseUnit)
+      .toFixed()
+  },
+)
+
+export const selectQuoteBuyAmountAfterFeesCryptoPrecision = createSelector(
+  selectQuoteBuyAmountAfterFeesBaseUnit,
+  (state: SwapperState) => state.buyAsset?.precision,
+  (quoteBuyAmountAfterFeesBaseUnit, buyAssetPrecision): string | undefined => {
+    if (!quoteBuyAmountAfterFeesBaseUnit || !buyAssetPrecision) return undefined
+    return fromBaseUnit(quoteBuyAmountAfterFeesBaseUnit, buyAssetPrecision)
+  },
+)
+
+export const selectQuoteBuyAmountAfterFeesFiat = createSelector(
+  selectQuoteBuyAmountAfterFeesCryptoPrecision,
+  selectBuyAssetFiatRate,
+  (quoteBuyAmountAfterFeesCryptoPrecision, buyAssetFiatRate): string | undefined => {
+    if (!quoteBuyAmountAfterFeesCryptoPrecision || !buyAssetFiatRate) return undefined
+    return bnOrZero(quoteBuyAmountAfterFeesCryptoPrecision).times(buyAssetFiatRate).toFixed()
+  },
+)
+
+export const selectTradeAmountsByActionAndAmountFromQuote: Selector<
+  SwapperState,
+  SelectTradeAmountsByActionAndAmountReturn
+> = createSelector(
+  selectBuyAmountBeforeFeesFiat,
+  selectSellAmountBeforeFeesFiat,
+  selectBuyAmountAfterFeesFiat,
+  selectBuyAmountAfterFeesCryptoPrecision,
+  selectSellAmountBeforeFeesCryptoPrecision,
+  selectBuyAmountBeforeFeesBuyAssetCryptoPrecision,
+  selectAction,
+  selectSellAsset,
+  selectBuyAsset,
+  selectAmount,
+  selectQuoteSellAmountPlusFeesBaseUnit,
+  selectQuoteSellAmountPlusFeesFiat,
+  selectQuoteBuyAmountAfterFeesCryptoPrecision,
+  selectQuoteBuyAmountAfterFeesFiat,
+  selectSlippage,
+  (
+    buyAmountBeforeFeesFiat,
+    sellAmountBeforeFeesFiat,
+    buyAmountAfterFeesFiat,
+    buyAmountAfterFeesCryptoPrecision,
+    sellAmountBeforeFeesCryptoPrecision,
+    buyAmountBeforeFeesCryptoPrecision,
+    action,
+    sellAsset,
+    buyAsset,
+    amount,
+    quoteSellAmountPlusFeesBaseUnit,
+    quoteSellAmountPlusFeesFiat,
+    quoteBuyAmountAfterFeesCryptoPrecision,
+    quoteBuyAmountAfterFeesFiat,
+    slippage,
+  ) => {
+    const defaultReturn: SelectTradeAmountsByActionAndAmountReturn = {
+      buyAmountBuyAssetCryptoPrecision: '0',
+      sellAmountSellAssetCryptoPrecision: '0',
+      fiatSellAmount: '0',
+      fiatBuyAmount: '0',
+    }
+    if (!sellAsset || !buyAsset || bnOrZero(amount).isZero()) return defaultReturn
+    const quoteBuyAmountAfterFeesCryptoPrecisionAfterSlippage = bnOrZero(
+      quoteBuyAmountAfterFeesCryptoPrecision,
+    )
+      .times(bn(1).minus(slippage))
+      .toFixed()
+    switch (action) {
+      case TradeAmountInputField.SELL_CRYPTO: {
+        return {
+          sellAmountSellAssetCryptoPrecision: amount,
+          buyAmountBuyAssetCryptoPrecision: quoteBuyAmountAfterFeesCryptoPrecisionAfterSlippage,
```

Comment:
semantic feedback only that @gomesalexandre will probably say the same about
```suggestion
          buyAmountBuyAssetCryptoPrecision: quoteBuyAmountAfterFeesAfterSlippageCryptoPrecision,
```

Reply:
https://github.com/shapeshift/web/pull/4217/commits/f2d17e969c3ee6b90ba2c6bc98fe623f3caf6fdc 👌